packages/ui/src/components/transactions_list/components/desktop/index.tsx

```diff
@@ -13,10 +13,41 @@ import Typography from '@mui/material/Typography';
 import useAppTranslation from '@/hooks/useAppTranslation';
 import Link from 'next/link';
 import numeral from 'numeral';
-import { FC, LegacyRef } from 'react';
+import React, { FC, MutableRefObject } from 'react';
 import AutoSizer from 'react-virtualized-auto-sizer';
 import { VariableSizeGrid as Grid } from 'react-window';
 import InfiniteLoader from 'react-window-infinite-loader';
+import FilterTxsByType from '@/components/transaction_type_filter';
+
+const useRenderHeaderCell = ({
+  columnIndex,
+  style,
+}: {
+  columnIndex: number;
+  style: React.CSSProperties;
+}) => {
+  const { key, align } = columns[columnIndex];
+  const isTypeKey = key === 'type';
+  const { t } = useAppTranslation('transactions');
+  const { classes } = useStyles();
+
+  return (
+    <div style={style} className={classes.cell}>
+      {isTypeKey ? (
+        <>
+          <Typography variant="h4" align={align}>
+            {t(key)}
+          </Typography>
+          <FilterTxsByType />
+        </>
+      ) : (
+        <Typography variant="h4" align={align}>
+          {t(key)}
+        </Typography>
+      )}
```

Comment:
These two blocks are almost the same, to avoid duplication I think I would just add the condition like this:

```typescript
{isTypeKey && <FilterTxsByType />}
```